app/src/main/java/ca/marshallasch/veil/controllers/RightMeshController.java

```diff
@@ -0,0 +1,290 @@
+package ca.marshallasch.veil.controllers;
+
+import android.content.Context;
+import android.content.Intent;
+import android.support.v4.content.LocalBroadcastManager;
+import android.util.Log;
+import android.widget.Toast;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+
+import java.io.Serializable;
+import java.util.Set;
+
+import ca.marshallasch.veil.DataStore;
+import ca.marshallasch.veil.proto.Sync;
+import io.left.rightmesh.android.AndroidMeshManager;
+import io.left.rightmesh.android.MeshService;
+import io.left.rightmesh.id.MeshId;
+import io.left.rightmesh.mesh.MeshManager;
+import io.left.rightmesh.mesh.MeshStateListener;
+import io.left.rightmesh.util.RightMeshException;
+
+import static io.left.rightmesh.mesh.MeshManager.DATA_RECEIVED;
+import static io.left.rightmesh.mesh.MeshManager.PEER_CHANGED;
+import static io.left.rightmesh.mesh.MeshManager.REMOVED;
+
+/**
+ *
+ * All RightMesh logic is abstracted into this class
+ *
+ * @author Weihan Li
+ * @version 1.0
+ * @created on 2018-07-09
+ * @name veil-droid
+ */
+
+public class RightMeshController implements MeshStateListener{
+    public static final String NEW_DATA_BROADCAST = "ca.marshallasch.veil.controllers.NEW_DATA_BROADCAST";
+    // MeshManager instance - interface to the mesh network.
+    AndroidMeshManager meshManager = null;
+    Context serviceContext = null;
+    public static final int DATA_PORT = 9182;
+
+    //MemoryStore instance - for storing data in local HashTable
+    DataStore dataStore = null;
+
+    private boolean meshActive = false;
```

Comment:
This is never actually used, I thought it might be but it isnt it can be removed.